Fix job cancellation on delete dataset

davelopez opened the pull request:
Fixes #13978

I hope this is the correct fix for the issue.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

davelopez committed:
Add API test for cancel job on deleted outputs

davelopez committed:
Fix job cancel on deleted output datasets

mvdbeek commented on lib/galaxy/managers/datasets.py:
```diff
@@ -263,6 +263,14 @@ def is_accessible(self, dataset_assoc, user, **kwargs):
+        Marks this dataset association as deleted.
+        """
+        super().delete(item, flush=flush)
+        self.stop_creating_job(item)
```
Puh ... I know we did this in the old history and it's a nice optimization, but it itches me that we update another resource here without being explicit about it (plus the actual operation is prone to races). Could we add a parameter here that controls whether or not we want to stop the associated job, that defaults to false in the API and true in the UI ? 

There is also an issue that was asking for the ability to stop the job once you delete one running output, an explicit parameter would fit well into that.

davelopez replied:
Sure! Thanks for the help!

Is 3c851fa, more or less, what you had in mind?

davelopez committed:
Add explicit parameter to stop job on deleted dataset

When deleting a dataset through the UI `stop_job` will be true, effectively cancelling the creating job if all its outputs are deleted.
By default the job will not be cancelled.

mvdbeek reviewed (approved):
Very nice, thank you!
